packager/launchEditor.js

```diff
@@ -30,19 +50,22 @@ function launchEditor(fileName, lineNumber) {
     return;
   }
 
-  var argument = fileName;
+  var editor = process.env.REACT_EDITOR || process.env.EDITOR;
+  var args = [fileName];
   if (lineNumber) {
-    argument += ':' + lineNumber;
+    args = getArgumentsForLineNumber(editor, fileName, lineNumber);
   }
 
-  var editor = process.env.REACT_EDITOR || process.env.EDITOR;
   if (editor) {
     console.log('Opening ' + chalk.underline(fileName) + ' with ' + chalk.bold(editor));
-    exec(editor + ' ' + argument, function(error) {
-      if (error) {
-        console.log(chalk.red(error.message));
-        printInstructions('How to fix');
+
+    var child = spawn(editor, args, {stdio: 'inherit'});
```

Comment:
`spawn` is kind of messy because you need to handle the `error` event, which may or may not prevent `exit` from getting emitted. If your reasoning behind `spawn` is to safely escape args, maybe try execFile which should work as long as the args don't need to be resolved by a subshell.